### PR TITLE
upgrade ASR-based CER eval for TTS

### PR DESCRIPTION
**Pull request by potato-inoue**
This PR is for evaluating the intelligibility performance of TTS.
Now **the ASR model cannot recognize the headword of sentences** because the synthesized speech hardly has the silence in the head. To solve this problem, I **added the silence into the head and the tail of the synthesized speech**. I found that **adding silence is useful for ASR**.

I did below:
- add silence into the head & tail of the speech (New w/ ' )
- remove ' (apostrophe)  to ignore the difference "xxx's" and "xxxs". e.g., "dogs" & "dog's" (New w/o ')

Now version
```
id: (lj050_0029)
Scores: (#C #S #D #I) 57 1 2 0
REF:  T H A t i s r e f l e c t e d i n d e f i n i t e a n d c o m p r e h e n s i v e o p e r a t i n g p r o c e d u r e s 
HYP:  * * I t i s r e f l e c t e d i n d e f i n i t e a n d c o m p r e h e n s i v e o p e r a t i n g p r o c e d u r e s 
Eval: D D S                                                                                                                   

id: (lj050_0030)
Scores: (#C #S #D #I) 24 0 3 0
REF:  T H E c o m m i s s i o n a l s o r e c o m m e n d s 
HYP:  * * * c o m m i s s i o n a l s o r e c o m m e n d s 
Eval: D D D                        
```

New version
```
id: (lj050_0029)
Scores: (#C #S #D #I) 60 0 0 0
REF:  t h a t i s r e f l e c t e d i n d e f i n i t e a n d c o m p r e h e n s i v e o p e r a t i n g p r o c e d u r e s 
HYP:  t h a t i s r e f l e c t e d i n d e f i n i t e a n d c o m p r e h e n s i v e o p e r a t i n g p r o c e d u r e s 
Eval:                                                                                                                         

id: (lj050_0030)
Scores: (#C #S #D #I) 27 0 0 0
REF:  t h e c o m m i s s i o n a l s o r e c o m m e n d s 
HYP:  t h e c o m m i s s i o n a l s o r e c o m m e n d s 
Eval:                                                            
```
CER

| Type | Set | Snt | Wrd | Corr | Sub | Del | Ins | Err | S.Err |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
| Now | dev | 250 | 22337 | 98.9 | 0.2 | 0.9 | 0.1 | 1.3 | 38.4 |
| New w/ ' | dev | 250 | 22337 | 99.7 | 0.2 | 0.1 | 0.1 | 0.5 | 21.6 |
| New w/o ' | dev | 250 | 22316 | 99.7 | 0.2 | 0.1 | 0.1 | 0.5 | 21.2 |
| Now |  eval | 250 | 21532 | 99.0 | 0.3 | 0.7 | 0.3 | 1.3 | 35.2 |
| New w/ ' | eval | 250 | 21532 | 99.5 | 0.2 | 0.2 | 0.2 | 0.7 | 22.8 |
| New w/o ' | eval | 250 | 21518 | 99.6 | 0.2 | 0.2 | 0.2 | 0.6 | 21.2 |

WER

| Type | Set | Snt | Wrd | Corr | Sub | Del | Ins | Err | S.Err |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
| Now | dev | 250 | 4218 | 96.5 | 2.3 | 1.2 | 1.4 | 4.9 | 43.6 |
| New w/ ' | dev | 250 | 4218 | 97.8 | 2.2 | 0.0 | 1.4 | 3.7 | 28.4 |
| New w/o ' | dev | 250 | 4218 | 97.8 | 2.1 | 0 | 1.4 | 3.6 | 28 |
| Now | eval | 250 | 4023 | 96.1 | 3.1 | 0.8 | 1.8 | 5.7 | 43.6 |
| New w/ ' | eval | 250 | 4023 | 97.1 | 2.8 | 0.1 | 1.8 | 4.7 | 35.6 |
| New w/o ' | eval | 250 | 4023 | 97.2 | 2.7 | 0.1 | 1.8 | 4.6 | 34 |

### CODE REVIEW

**Comment by potato-inoue** [comment]
Now the scripts generate the wave file added silence as the intermediate files.
If you know the clever way not to generate the intermediate files, please let me know.

**Comment by potato-inoue** [comment]
The results of the top comment indicate the results of ground-truth. Other results (New w/o ') of pre-trained models are in `RESULT.md`. These results were evaluated by the samples uploaded on Google Drive.

**Comment by sw005320** [comment]
We can do it with `sox`, and also we can write the sox command in `wav.scp`. What do you think?

**Comment by kan-bayashi** [comment]
I agree with Shinji’s idea.

**Comment by potato-inoue** [comment]
Thanks, I try to connect silence waveform into the head of synthesized speech by using `sox` command.

**Comment by kan-bayashi** [comment]
@potato-inoue Could you re-evaluate samples in v.0.6.0 directory?
I added the ASR results but it is better to update.

**Comment by kan-bayashi** [comment]
@potato-inoue How is the progress?

**Comment by potato-inoue** [comment]
Now I'm making the scripts of evaluating other recipes, especially English recipes.
I will show these results by 12/19.

**Comment by potato-inoue** [comment]
These are results of CER. Results are sorted at the order that is from high to low at CER.

ljspeech_dev


ljspeech_eval

File: egs/ljspeech/tts1/local/ob_eval/data_prep_for_asr.sh [hunk 1]
```diff
@@ -27,9 +27,11 @@ text=${data_dir}/text
 [ -e ${utt2spk} ] && rm ${utt2spk}
 
 # make scp, utt2spk, and spk2utt
+sox=`which sox` || { echo "Could not find sox in PATH"; exit 1; }
+silence="local/ob_eval/silence_16k_100ms.wav"
```
Comment (sw005320) [inline egs/ljspeech/tts1/local/ob_eval/data_prep_for_asr.sh]: how did you get this silence?
Could you add some comments?

**Comment by sw005320** [comment]
Any progress?

File: egs/README.md [hunk 2]
```diff
@@ -5,6 +5,54 @@ See: https://espnet.github.io/espnet/tutorial.html
 
 ## Overview of example information
 
+<<<<<<< HEAD
```
Comment (sw005320) [inline egs/README.md:8]: Please fix the conflict.

File: egs/an4/asr1/local/ob_eval/evaluate_cer.sh [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+fi
+if [ ${eval_tts_model} == true ]; then
+    echo "Evaluate: TTS model"
+else
+    echo "Evaluate: ground truth"
+    expdir=exp/ground_truth
+    outdir=${expdir}/sym_link
+
+    mkdir -p ${outdir}_denorm/${name}/wav
+    # cat < data/${name}/wav.scp | awk '{print $1}' | while read -r wav_id; do
```
Comment (sw005320) [inline egs/an4/asr1/local/ob_eval/evaluate_cer.sh:65]: remove comment-out lines.

**Review by sw005320 — COMMENTED** [review]
Welcome back, @potato-inoue
- `ob_eval` is not clear. Maybe `asr_eval`?
-  It's better to move ASR evaluation related configs to `egs/*/tts1/conf/asr_eval`. If you put them under the `conf` directory, people may confuse

@kan-bayashi, could you also review this PR?

I think we should also have this in espnet2.
Actually, espnet2 would make it very easy.

**Pull request closed without merging**